packages/material-ui/src/MenuList/MenuList.js

```diff
@@ -105,56 +134,52 @@ class MenuList extends React.Component {
     const currentFocusIndex = items.indexOf(currentFocus);
 
     if (currentFocusIndex !== -1) {
-      return this.setTabIndex(currentFocusIndex);
+      return setCurrentTabIndex(currentFocusIndex);
     }
 
-    if (this.selectedItemRef) {
-      return this.setTabIndex(items.indexOf(this.selectedItemRef));
+    if (selectedItemRef.current) {
+      return setCurrentTabIndex(items.indexOf(selectedItemRef.current));
     }
 
-    return this.setTabIndex(0);
+    return setCurrentTabIndex(0);
   }
 
-  render() {
-    const { children, className, onBlur, onKeyDown, disableListWrap, ...other } = this.props;
-
-    return (
-      <List
-        role="menu"
-        ref={ref => {
-          this.listRef = ReactDOM.findDOMNode(ref);
-        }}
-        className={className}
-        onKeyDown={this.handleKeyDown}
-        onBlur={this.handleBlur}
-        {...other}
-      >
-        {React.Children.map(children, (child, index) => {
-          if (!React.isValidElement(child)) {
-            return null;
-          }
-
-          warning(
-            child.type !== React.Fragment,
-            [
-              "Material-UI: the MenuList component doesn't accept a Fragment as a child.",
-              'Consider providing an array instead.',
-            ].join('\n'),
-          );
-
-          return React.cloneElement(child, {
-            tabIndex: index === this.state.currentTabIndex ? 0 : -1,
-            ref: child.props.selected
-              ? ref => {
-                  this.selectedItemRef = ReactDOM.findDOMNode(ref);
-                }
-              : undefined,
-            onFocus: this.handleItemFocus,
-          });
-        })}
-      </List>
-    );
-  }
+  return (
+    <List
+      role="menu"
+      ref={ref => {
+        listRef.current = ReactDOM.findDOMNode(ref);
```

Comment:
For every `findDOMNode` call add `// #StrictMode ready` if the `ref` is either attached to a host component or to a ref forwarding component or `// not #StrictMode ready` if not.

Reply:
I don't understand your point. What about?
```suggestion
    <List
      role="menu"
      ref={listRef}
```

Reply:
Please explain first why your approach is better.

Reply:
Now that `forwardRef` is being used in `withStyles` and `List`, it shouldn't be necessary to retain the `findDOMNode` call at all and getting rid of it would make it more clearly "StrictMode ready".

Reply:
That was aimed @oliviertassinari. The `findDOMNode` should still be used for backwards compat with class components e.g. `<List component={SomeClassComponent } ref={notADOMNodeRef} />`

 I explained this multiple times to the team already which is why I asked what would be different here.

Reply:
@eps1lon Is there a path in mind to eventually being able to drop the use of `findDOMNode` in this case?

Reply:
@eps1lon I get it now. The comment is about documenting the usages of `findDOMNode()` in the code, to keep track of the `findDOMNode()` calls that raises a warning in StrictMode. This way, we can keep track of our progress. Thank you for taking the time. I hope we can. 

@ryancogswell We are waiting on the React core team feedback on https://github.com/facebook/react/pull/15091. Whether or not we can keep using `findDOMNode()` like this. The long term vision is tracked in #14761.

Reply:
I guess we can drop it for v5. It's used for compatiblity with the rest of the ecosystem only.

Reply:
This is just for my own reference to understand the terminology in this discussion:
https://reactjs.org/docs/implementation-notes.html#mounting-host-elements

I wasn't clear on the meaning of "attached to a **host** component", but now I think I am.

@eps1lon Would you consider this case to be "// #StrictMode ready" since the ref would be attached to a ref forwarding component (i.e. `List`)? Or is it "// not #StrictMode ready" because the `List` component is in turn attaching the ref to a component that might not meet the criteria. I'm guessing the former (indicating that the only reason `findDOMNode` is still present is because of issues deeper in another component), because if this would be the marked as "// not #StrictMode ready" then I wouldn't expect there to be any "// #StrictMode ready" cases where you couldn't go ahead and remove the `findDOMNode` call.

Reply:
I would consider this strict mode ready. I view strict mode ready for any code we have or depend on. Basically if you use material-ui in your application you should be able to make it strict mode ready. This is achievable by migrating to `forwardRef` for components passed into `component` props in your application.

If we would use e.g. `findDOMNode(this)` then users of our library have no chance of making their application strict mode ready because our components aren't strict mode ready. Hope that makes sense.

Thanks for the docs link. I switched back and forth with the terminology for "string components" (intrinsic, native, host etc). The docs confirm that "host component" is the correct term.

Reply:
@oliviertassinari Thanks for the additional context. If this particular case would be marked as "// #StrictMode ready", then I think I understand. If not, then I'm confused on some of the details.

Reply:
@ryancogswell Correct. To summarize: `findDOMNode` calls are 
- strict mode ready when the argument is a ref that is attached to a component that forwards that ref to a host component by **default** (essentially making `findDOMNode` unnecessary; it would just be the identity function)
- **not** strict mode ready when the argument is a ref that is attached to a component that does not forward its ref to a host component

Reply:
@eps1lon Sorry to ask for more clarification on this, but there is still some gray area in this for me.

For this case:
```
ref: child.props.selected
            ? ref => {
                selectedItemRef.current = ReactDOM.findDOMNode(ref);
              }
            : undefined,
```
The typical case would be for `ref` to be a `MenuItem` which would fit the "strict mode ready" case, but unlike the `List` case where we know it is going to `List` (we just don't know whether or not the user may have specified a custom component in the List props), here we are just looping over children for which we have no direct control over the component type. I'm assuming this is a  "// not #StrictMode ready" case?